src/Data/Vector/Mutable/Linear.hs

```diff
@@ -237,7 +237,7 @@ mapMaybe vec (f :: a -> Maybe b) =
   go r w s vec'
     -- If we processed all elements, set the capacity after the last written
     -- index and coerce the result to the correct type.
-    | r == s =
+    | r Prelude.== s =
```

Comment:
I'm questioning why all these `==` have been replaced by `Prelude.==`, the linear one is more general.

Reply:
I think that was because after we merged https://github.com/tweag/linear-base/pull/209, we don't have many `Eq` instances we could use (however, it seems like it has an instance of `Int`, I'll double check). I have a PR in the making which adds some more instances, after that we can use our `==` more often, I hope.

Reply:
They were unnecessary as you said, I started using linear versions.